pkg/destroy/gcp/gcp.go

```diff
@@ -97,7 +95,7 @@ func (o *ClusterUninstaller) Run() (*types.ClusterQuota, error) {
 
 	o.cpusByMachineType = map[string]int64{}
 	req := o.computeSvc.MachineTypes.AggregatedList(o.ProjectID).Fields("items/*/machineTypes(name,guestCpus),nextPageToken")
-	if err := req.Pages(o.Context, func(list *compute.MachineTypeAggregatedList) error {
+	if err := req.Pages(ctx, func(list *compute.MachineTypeAggregatedList) error {
```

Comment:
This is going from a never-expiring context to a 2-minute-timeout context in which some time already passed. Since this is caching machine types by iterating through possible pages of data, maybe we should use a longer timeout context? Or maybe we create a child context with longer timeout just for this?